packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -472,16 +472,14 @@ class InputDecorator extends StatelessWidget {
 
     final Color activeColor = _getActiveColor(themeData);
 
-    final TextStyle baseStyle = this.baseStyle ?? themeData.textTheme.subhead;
+    final TextStyle baseStyle = themeData.textTheme.subhead.merge(this.baseStyle);
```

Comment:
Have you checked if `floatingLabelStyle.fontSize` can hit the same issue?

Reply:
Yes, fixed.